fix(rides): filter out future rides from gap list

TzivyaFireman opened the pull request:
# Description

Filter out future rides from the gap list to prevent rides scheduled later the same day from being shown as missing.
Also applies the same filtering to the overall gap percentage calculation.

## screenshots
❌
<img width="452" height="657" alt="image" src="https://github.com/user-attachments/assets/0f660b5c-399e-4e9e-b5ad-077d493e45df" />
 ✅
<img width="393" height="413" alt="image" src="https://github.com/user-attachments/assets/78c4b889-bb8c-4907-b931-3df93f6b3111" />

NoamGaash commented:
I thought that we planned to show those rides, with an annotation mentioning that those rides are in the future and therefore we can't know in advanced if it will be executed or not. Is it better to just hide it?

AvivAbachi commented:
Hi @TzivyaFireman,

The pull request looks great, but a few changes are needed:

- Instead of filtering the `gaps`, please add a new status in `formatStatus`.
- You need to update your branch by merging from the `main` branch, because it is outdated. 
  - Change `moment()` to `dayjs()`.
  - Reverted the changes in `SideBar.tsx`.
  > After merging, please run `npm install` to update your dependencies.

Thank you!

TzivyaFireman commented:
**Hi,
I made some significant changes**

- Added a new status: "ride_in_future"
- Added appropriate translations in the i18n json files
- Future rides are excluded from the ride gap percentage calculation
- In the "ride gaps only" view, future rides are completely hidden
To keep the codebase and commit history clean, I created a new branch that starts from a more up-to-date version of the code
<img width="417" height="460" alt="image" src="https://github.com/user-attachments/assets/a307d721-70e0-48c9-9725-4852b1090322" />

NoamGaash commented:
Looks amazing! I'll review the code in the evening. Great job :clap:

NoamGaash reviewed (approved):
Perfect! Thank you :clap: 

If you want - would you make another PR with Storybook update to demonstrate this "future ride" option?

TzivyaFireman commented:
I don’t have experience with Storybook, but I can give it a try.
If I see it’s taking too long, should I open a new issue for this?

AvivAbachi commented on src/pages/gaps/index.tsx:
```diff
@@ -42,6 +42,10 @@ const GapsPage = () => {
   }
 
   function formatStatus(all: GapsList, gap: Gap) {
+    const gapTime = gap.gtfsTime || gap.siriTime
```
**Some Team Notes:**

- **GTFS:** Scheduled Rides Data  
- **SIRI:** GPS Data  

**Suggestions:**  
1. Use only **GTFS** Time 
2. Consider edge cases for early rides where the **GTFS** schedule indicates a time after the current time, but **SIRI** GPS time is available.
3. Add a buffer of a few minutes, since data updates occur every 3-5 minutes.

AvivAbachi commented on src/pages/gaps/index.tsx:
```diff
@@ -56,9 +60,16 @@ const GapsPage = () => {
+    if (!gaps) return undefined
+    const relevantGaps = gaps.filter((gap) => {
+      const gapTime = gap.gtfsTime || gap.siriTime
+      return gapTime && gapTime.isBefore(dayjs())
```
1. Use GTFS time with the same time buffer if added.
 2. Instead, create a new `dayjs()` object each time during looping using: 
```js
const now = dayjs();
```

AvivAbachi reviewed (commented):
Looking good, nice pr.

TzivyaFireman commented:
Added an [issue](https://github.com/hasadna/open-bus-map-search/issues/1202),
Should I wait before merging this PR until the Storybook is updated?

AvivAbachi commented:
@TzivyaFireman 
Noam and I fixed some of the random failing tests.
You can merge now.
